modules/ppcp-settings/docs/merchant-post-onboarding.md

```diff
@@ -0,0 +1,414 @@
+    PG->>M: Redirect to settings page
+```
+
+## Merchant Data Sources
+
+### PayPal OAuth Response
+
+When the merchant completes PayPal OAuth authentication, PayPal returns these key data points:
+
+**File:** `modules/ppcp-settings/src/Handler/ConnectionListener.php`
+
+```php
+// OAuth response parameters
+'merchantIdInPayPal' => 'ANONYMOUS_MERCHANT_ID',  // PayPal's internal merchant ID
+'merchantId'         => 'merchant@email.com',     // Merchant's PayPal email address
+'accountStatus'      => 'BUSINESS_ACCOUNT',       // Account type indicator
+'ppcpToken'          => 'VALIDATION_TOKEN'        // Security token
+```
+
+#### Data Extraction Process
+
+The `ConnectionListener` extracts and sanitizes merchant information:
+
+1. **Merchant ID**: Anonymous PayPal merchant identifier (`merchantIdInPayPal`)
+2. **Merchant Email**: PayPal account email address (`merchantId` parameter)
+3. **Seller Type**: Determined from `accountStatus` parameter
+   - `'BUSINESS_ACCOUNT'` → `SellerTypeEnum::BUSINESS`
+   - Empty/missing → `SellerTypeEnum::PERSONAL`
+   - Unknown values → `SellerTypeEnum::UNKNOWN`
+
+### Merchant Country Determination
+
+The merchant's operating country comes from PayPal's API after successful authentication:
+
+**File:** `modules/ppcp-wc-gateway/src/Helper/MerchantDetails.php`
+
+```php
+/**
+ * The merchant's country according to PayPal, which might be different from
+ * the WooCommerce country.
+ */
+private string $merchant_country;
+
+/**
+ * Returns the merchant's country. This country is used by PayPal to decide
+ * which features the merchant can access.
+ */
+public function get_merchant_country(): string {
+    return $this->merchant_country;
+}
+```
+
+#### Country Sources Priority
+
+1. **PayPal API Response**: Primary source - merchant's registered country with PayPal
+2. **Store Country Fallback**: WooCommerce store country setting if PayPal data unavailable
+3. **Feature Determination**: PayPal merchant country determines available features
+
+**Service Registration:**
+```php
+// modules/ppcp-settings/services.php
+'settings.merchant-details' => function(ContainerInterface $container): MerchantDetails {
+    $data = $container->get('settings.data.general');
+    $merchant_country = $data->get_merchant_country();  // From PayPal API
+    $woo_data = $data->get_woo_settings();
+
+    return new MerchantDetails($merchant_country, $woo_data['country'], $eligibility_checks);
+},
+```
+
+## Post-Connection System Changes
+
+### 1. Connection State Management
+
+**File:** `modules/ppcp-wc-gateway/src/Helper/ConnectionState.php`
+
+The system maintains connection state that affects all plugin behavior:
+
+```php
+class ConnectionState {
+    private bool $is_connected;
+    private Environment $environment;
+
+    public function is_connected(): bool {
+        return $this->is_connected;
+    }
+
+    public function is_sandbox(): bool {
+        return $this->environment->is_sandbox();
+    }
+}
+```
```

Comment:
Suggestion: Instead of documenting class code, IMO it would be beneficial to document the service name + the API methods (public methods) - but actually this section here is already 100% covered by the doc-block comment in the `.php` file and only duplicates content.

https://github.com/woocommerce/woocommerce-paypal-payments/blob/060a86f5c9693872328e4fbb9f58718e8456d091/modules/ppcp-wc-gateway/src/Helper/ConnectionState.php#L12-L20 

A link-list to all relevant classes would give the reader a better overview of what's involved in the post-onboarding process.